packages/cardano-services/src/Program/services/postgres.ts

```diff
@@ -82,11 +82,13 @@ const mergeTlsOptions = (
 
 export const getConnectionConfig = (
   dnsResolver: DnsResolver,
-  options?: PosgresProgramOptions
+  options?: PosgresProgramOptions<'Projection'>
```

Comment:
If here we declare 'Projection' scope always then we should rename the function as well that works only for projection connection config, otherwise could be a generic one working with all types of config

Reply:
Is there a case where `getConnectionConfig` could be reused to provide config for all databases based on the input? If yes - then generic util would be more useful

Reply:
The only function which currently needs to be generic (i.e. used with both the suffixes) is `connectionStringFromArgs`.

I tried to avoid overengineering: when in the future some other function will need to be refactored being generic we can do that.

Another point is I thought that applying changes not strictly required (as making generic a function which doesn't need to be changed) is not a good time investment.